Odd error installing xdebug while building an 8.2 container in gitlab CI

### Version of install-php-extensions

v.1.5.51

### Error description

When attempting to install xdebug in php:8.2 while building a container in gitlab CI I'm getting an odd error:
```
2: /tmp/pickle.tmp/xdebug/xdebug-3.2.0/src/lib/usefulstuff.c:1153: error: unterminated #else
2:  1153 | #ifdef ZEND_HASH_GET_APPLY_COUNT /* PHP 7.2 or earlier recursion protection */
2:       |
2: make: *** [Makefile:270: src/lib/usefulstuff.lo] Error 1The command '/bin/sh -c /usr/local/bin/install-php-extensions ${EXTENSIONS} @composer' returned a non-zero code: 1
```

The full output can be seen here: https://gitlab.com/portlandlabs/containers/-/jobs/3479191268

For some reason I'm able to build this container without issue locally, this issue only shows up in gitlab CI.

### Docker image

php:8.2

### Minimal Dockerfile

```Dockerfile
https://gitlab.com/portlandlabs/containers/-/blob/main/php/default/Dockerfile

where:
- `BASE_IMAGE` = `php:8.2`
- `EXTENSIONS` = `mbstring curl intl gd xml zip bz2 pdo_mysql xdebug redis amqp`
```

One odd thing about this is the errors seem to point to lines of code that don't exist in certain files. One fatal error is directly related to this:
```2: /tmp/pickle.tmp/xdebug/xdebug-3.2.0/src/base/filter.c:346:10: fatal error: mm.h: No such file or directory
2:   346 | #include "mm.h"
2:       |          ^~~~~~
2: compilation terminated.
```
Note that the line it's complaining about isn't actually in `base/filter.c` instead it's in `lib/str.c` so the relative include `#include "mm.h"` ends up fataling since it's relative to `base` rather than the intended `lib` directory.

--- a/install-php-extensions
+++ b/install-php-extensions
@@ -3665,14 +3665,13 @@ moduleMayUsePecl() {
 # Updates:
 #   PHP_MODULES_TO_INSTALL
 # Sets:
-#   USE_PICKLE
+#   USE_PICKLE 0: no, 1: yes (already downloaded), 2: yes (build it from source)
 configureInstaller() {
 	USE_PICKLE=0
-	for PHP_MODULE_TO_INSTALL in $PHP_MODULES_TO_INSTALL; do
-		if moduleMayUsePecl "$PHP_MODULE_TO_INSTALL"; then
-			if test $PHP_MAJMIN_VERSION -lt 800; then
-				pecl channel-update pecl.php.net || true
-				return
+	if ! which pecl >/dev/null; then
+		for PHP_MODULE_TO_INSTALL in $PHP_MODULES_TO_INSTALL; do
+			if ! moduleMayUsePecl "$PHP_MODULE_TO_INSTALL"; then
+				continue
 			fi
 			if false && anyStringInList '' "$PHP_MODULES_TO_INSTALL"; then
 				USE_PICKLE=2
@@ -3681,9 +3680,12 @@ configureInstaller() {
 				chmod +x /tmp/pickle
 				USE_PICKLE=1
 			fi
-			return
-		fi
-	done
+			break
+		done
+	fi
+	if test $USE_PICKLE -eq 0; then
+		pecl channel-update pecl.php.net || true
+	fi
 }
 
 buildPickle() {

